### ISSUE TITLE
Changing F1 to ROC

### ISSUE THREAD

**Issue by apavlo89** [8080labs/ppscore#43]
Hello,

First of all PPScore is so good, great job! Running only PPS score to reduce feature numbers gives better results than using other feature elimination techniques or running it first before using other feature elimination techniques ALWAYS yields better results than running pretty much every other feature elimination technique(s) alone.

I just wanted to ask how to change from F1 score to ROC, specifically, Precision-Recall Curves (as I have moderate imbalanced classes), for my binary classification problem.

Thank you for your help in the matter.

Regards,

Achilleas

**Comment by FlorianWetschoreck** [comment]
Hi Achilleas,

thank you for your feedback and happy to hear that ppscore is helpful to you.
Currently, you cannot change the classification score from F1 to ROC but I agree that this would be a useful addition.
How would you like to change the score from F1 to ROC? How would you like the user experience/API to look like?

Regards,
Florian

**Comment by FlorianWetschoreck** [comment]
I like the suggestion and think that this can be worthwhile.
About the API proposal:
- we might need to find a way to differentiate between regression and classification metrics (in case of supporting this within predictors)
- regarding the `cross_validation=LOOCV_method` was it on purpose that you refer to the LOOCV_method object that might be defined somewhere or did you want to refer to it via a string?
- About adding other evaluation metrics: for all of them we need to define the limits in terms of best case and bottom baseline. In some cases like the F1 this might not always be so trivial

Also, would you like to take on this task?

**Comment by apavlo89** [comment]
1) Cool.
2) No I defined LOOCV_method before
3) Yes, I agree. Choosing the right performance metric depends on the data.

I wish I could but I don't have the technical knowledge to do this I'm afraid!

**Comment by FlorianWetschoreck** [comment]
Alright, thank you for the answers.
And no worries if you cannot work on it yourself - the suggestion and rough desgn of the user experience is also very valuable.
Also, maybe you will rise to the occasion at some point - after all, we grow with our challenges :)

Take care and talk to you soon

**Comment by dibble07** [comment]
I would also like a better approach to handling class imbalance and am happy to have a go at adding this functionality. I was thinking of making the setting of the `average` parameter in the [f1_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score) as a user optional input instead of as a hardcoded value of "weighted". This would be a pretty lightweight approach and give you all the functionality and error handling of the underlying sklearn function. How does this sound?

**Comment by dibble07** [comment]
I would also like a better approach to handling class imbalance and am happy to have a go at adding this functionality. I was thinking of making the setting of the `average` parameter in the [f1_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score) as a user optional input instead of as a hardcoded value of "weighted". This would be a pretty lightweight approach and give you all the functionality and error handling of the underlying sklearn function. How does this sound @FlorianWetschoreck ?
